### PR TITLE
Fixed not switching Disassembly to not opened Graph view using <space> + a bit refactor

### LINKED ISSUE #1268: Pressing <space> in Disassembly view should open Graph if closed (and vice versa)
**Is your feature request related to a problem? Please describe.**
When the user presses the <kbd>space</kbd> key in the Disassembly widget and the Graph widget is closed, nothing happens. Same for when clicking <kbd>space</kbd> in Graph when Disassembly widget is closed.

**Describe the solution you'd like**
If <kbd>space</kbd> was clicked in either disassembly or graph widgets, the other widget should be opened by Cutter if it is closed.

### PR DESCRIPTION
**Pull request by AntonReborn**
### What had to be fixed: ###
Fixes #1268 

### What has been changed: ###
Moved logic behind the handling of the `raisePrioritizedMemoryWidget` signal to the common base class `MemoryDockWidget`. It triggers an action corresponding to the memory widget (Disassembly, Graph, Hexdump, Pseudocode), shows, raises and set focus to it.

### Test plan: ###

- Close Graph dock window;
- Go to disassembly;
- Press space a few times;
- The Graph view should be opened and switching should happen.

### CODE REVIEW

File: src/widgets/GraphWidget.h [hunk 1]
```diff
@@ -1,25 +1,32 @@
 #ifndef GRAPHWIDGET_H
 #define GRAPHWIDGET_H
 
-#include "CutterDockWidget.h"
+#include "MemoryDockWidget.h"
+#include "core/Cutter.h"
```
Comment (xarkes) [inline src/widgets/GraphWidget.h]: Why do you need to include `Cutter.h` here?

Comment (AntonReborn) [reply src/widgets/GraphWidget.h]: Don't need, thanks

**Comment by Vane11ope** [comment]
overall looks pretty good to me too especially those refactoring such as making the properties private and using getter to access them instead of touching directly. 

just one styling issue to keep our coding conventions and i'm good.

File: src/widgets/MemoryDockWidget.cpp [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    : CutterDockWidget(parent, action)
+    , mType(type)
+{
+    connect(Core(), &CutterCore::raisePrioritizedMemoryWidget, this, &MemoryDockWidget::handleRaiseMemoryWidget);
+}
+
+void MemoryDockWidget::handleRaiseMemoryWidget(CutterCore::MemoryWidgetType raiseType)
+{
+    bool raisingEmptyGraph = (raiseType == CutterCore::MemoryWidgetType::Graph && Core()->isGraphEmpty());
+    if (raisingEmptyGraph)
```
Comment (Vane11ope) [inline src/widgets/MemoryDockWidget.cpp]: please put the curly braces just for our coding conventions.

Comment (AntonReborn) [reply src/widgets/MemoryDockWidget.cpp]: Cutter uses [Qt Code Style](https://wiki.qt.io/Qt_Coding_Style). 

> Use curly braces only when the body of a conditional statement contains more than one line:

Comment (ITAYC0HEN) [reply src/widgets/MemoryDockWidget.cpp]: We will update our [docs](Cutter.re) to specify this coding style. Thank you for pointing it out :)

Comment (Vane11ope) [reply src/widgets/MemoryDockWidget.cpp]: @ITAYC0HEN update how? i mean if we really think of unifying the coding conventions, then I think it'd rather be not reasonable to ask to remove the curly braces for that condition every time, and it'd be way easier for people to remember use curly braces in any case than to remember if (one line) { then not use curly braces } imho.

Comment (Vane11ope) [reply src/widgets/MemoryDockWidget.cpp]: but you know what, in fact i can't care less about such a super tiny point. so i resolve this. good PR anyway, feel free to merge @ITAYC0HEN after the conflicts being solved :)

**Comment by AntonReborn** [comment]
@xarkes fixed a performance issue in the last commit. I do not know all the invariants in a graph view update flow. But this refresh looks redundant and seems like I didn't break anything.

**Comment by xarkes** [comment]
Yep much better, there's just a tiny issue that remains ...
I don't know if it's important to anyone, but if you scroll the graph, and press space, the position in the graph is not saved.
If no one cares about it, we can merge.

**Comment by xarkes** [comment]
Thanks! :-)

**Pull request merged**